internal/generateOmgIdl.test.ts

```diff
@@ -0,0 +1,183 @@
+import { parseIdlToMessageDefinition } from "@foxglove/omgidl-parser";
+
+import { DURATION_IDL, TIME_IDL, generateOmgIdl } from "./generateOmgIdl";
+import { foxgloveEnumSchemas, foxgloveMessageSchemas } from "./schemas";
+import { exampleEnum, exampleMessage } from "./testFixtures";
+
+describe("generateOmgIdl", () => {
+  it("generates .idl files", () => {
+    expect(generateOmgIdl(exampleEnum)).toMatchInlineSnapshot(`
+      "// Generated by https://github.com/foxglove/schemas
+
+      module foxglove {
+
+      // An example enum
+      enum ExampleEnum {
+        // Value A
+        // Value: 0
+        A,
+
+        // Value B
+        // Value: 1
+        B
+      };
+
+      };
+      "
+    `);
+    expect(generateOmgIdl(exampleMessage)).toMatchInlineSnapshot(`
+      "// Generated by https://github.com/foxglove/schemas
+
+      #include "foxglove/Duration.idl"
+      #include "foxglove/ExampleEnum.idl"
+      #include "foxglove/NestedMessage.idl"
+      #include "foxglove/Time.idl"
+
+      module foxglove {
+
+      // An example type
+      struct ExampleMessage {
+        // duration field
+        Duration field_duration;
+
+        // time field
+        Time field_time;
+
+        // boolean field
+        boolean field_boolean;
+
+        // bytes field
+        sequence<uint8> field_bytes;
+
+        // float64 field
+        double field_float64;
+
+        // uint32 field
+        uint32 field_uint32;
```

Comment:
You could support default values for primitives (except string) with `@default(value=<val>)`. not sure if this was an intentional omission.

Reply:
not intentional, I just forgot/didn't realize that is a thing 😄